frappe/public/js/frappe/form/controls/data.js

```diff
@@ -148,6 +149,37 @@ frappe.ui.form.ControlData = class ControlData extends frappe.ui.form.ControlInp
 				}
 			});
 		});
+
+		this.$wrapper.find('.control-input').append(
+			`<span class="link-btn scan-settings" style="display: inline;">
+				<a class="btn-open no-decoration" title="${__("Settings")}">
+					${frappe.utils.icon('setting-gear', 'md')}
+				</a>
+			</span>`
+		);
+		this.$scan_settings_btn = this.$wrapper.find('.scan-settings');
+		this.$scan_settings_btn.on('click', 'a', () => {
+			const settings = frappe.get_user_settings(me.frm.doc.doctype, "BarcodeScan");
+
+			new frappe.ui.Dialog({
+				title: __("Update Barcode Scan Settings"),
+				fields: [
+					{
+						fieldtype: "Check",
+						fieldname: "prompt_qty",
+						label: __("Prompt Qty"),
```

Comment:
Framework's barcode code is ONLY for adding button to enable scanning using device's camera. This change/config is _VERY_ specific to ERPNext, lets keep this ERPNext?